ament_cmake_auto/cmake/ament_auto_find_build_dependencies.cmake

```diff
@@ -27,18 +32,35 @@
 # @public
 #
 macro(ament_auto_find_build_dependencies)
-  if(ARGN)
+  cmake_parse_arguments(ARG "" "" "REQUIRED_PACKAGES" ${ARGN})
```

Comment:
In a macro all variables should be prefixed with an underscore to avoid overwriting external variables: `_ARG`

All following derived variable names need to be updated accordingly.